### PR TITLE
tagmap layer example improvements

### PR DESCRIPTION
**Pull request by Pessimistress**
For https://github.com/uber/deck.gl/issues/1619

#### Change List
- Remove unnecessary dependencies
- Perf: add caching for layout results (5fps -> 40fps)
- Misc. code clean up

### CODE REVIEW

File: examples/tagmap/tagmap-layer/tagmap-wrapper.js [hunk 1]
```diff
@@ -2,78 +2,61 @@
...
     this.tagMap.buildHierarchy(data, {getLabel, getPosition, getWeight});
   }
 
-  setVisParam({minFontSize, maxFontSize, weightThreshold, colorScheme}) {
-    this.visParam = {minFontSize, maxFontSize, weightThreshold, colorScheme};
+  setVisParam({minFontSize, maxFontSize, weightThreshold}) {
+    this.visParam = {minFontSize, maxFontSize, weightThreshold};
   }
 
   getTags({transform, viewport}) {
```
Comment (gnavvy) [inline examples/tagmap/tagmap-layer/tagmap-wrapper.js]: no need for *viewport* if we remove the bbox computation.

**Comment by gnavvy** [comment]
@rivulet-zhang could you also take a look?

File: examples/tagmap/app.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ class Root extends Component {
 
     requestJson(FILE_PATH, (error, response) => {
```
Comment (ibgreen) [inline examples/tagmap/app.js]: For another PR: we should update examples to use `fetch` so we can drop another dependency...

File: examples/tagmap/tagmap-layer/tagmap-layer.js [hunk 3]
```diff
@@ -28,52 +37,73 @@ export default class TagmapLayer extends CompositeLayer {
...
-    const {tagMap} = this.state;
+    const {tagMap, tagsCache} = this.state;
     if (!tagMap) {
       return;
     }
 
     const {viewport} = this.context;
-    const {minFontSize, maxFontSize, weightThreshold, colorScheme} = this.props;
-    const transform = new WebMercatorViewport(Object.assign({}, viewport));
+    const z = Math.floor(viewport.zoom);
```
Comment (ibgreen) [inline examples/tagmap/tagmap-layer/tagmap-layer.js]: Nit: We don't use one letter variables... How about `discreteZoomLevel`

File: examples/tagmap/webpack.config.js [hunk 4]
```diff
@@ -48,4 +48,4 @@ module.exports = {
 
 // DELETE THIS LINE WHEN COPYING THIS EXAMPLE FOLDER OUTSIDE OF DECK.GL
 // It enables bundling against src in this repo rather than installed deck.gl module
-// module.exports = require('../webpack.config.local')(module.exports);
+module.exports = require('../webpack.config.local')(module.exports);
```
Comment (ibgreen) [inline examples/tagmap/webpack.config.js]: Nit: There is usually an `env` check on this line to make this work outside of repo?

File: examples/text-layer/webpack.config.js [hunk 5]
```diff
@@ -48,4 +48,4 @@ module.exports = {
 
 // DELETE THIS LINE WHEN COPYING THIS EXAMPLE FOLDER OUTSIDE OF DECK.GL
 // It enables bundling against src in this repo rather than installed deck.gl module
-// module.exports = require('../webpack.config.local')(module.exports);
+module.exports = require('../webpack.config.local')(module.exports);
```
Comment (ibgreen) [inline examples/text-layer/webpack.config.js]: Ditto

**Pull request merged**